aiida_codtools/cli/misc/cif_import.py

```diff
@@ -164,22 +168,26 @@ def launch(group, database, max_entries, number_species, skip_partial_occupancie
         source_id = entry.source['id']
 
         if source_id in existing_cif_nodes:
-            click.echo('{} | Cif<{}> skipped: already present in group {}'.format(
+            click.echo('{} | Cif<{}> skipping: already present in group {}'.format(
                 datetime.utcnow().isoformat(), source_id, group.name))
             continue
 
         try:
             cif = entry.get_cif_node()
-        except (StarError, HTTPError, UnicodeDecodeError) as exception:
-            click.echo('{} | Cif<{}> skipped: encountered an error retrieving cif data: {}'.format(
-                datetime.utcnow().isoformat(), source_id, exception))
+        except (AttributeError, UnicodeDecodeError, StarError, HTTPError) as exception:
+            click.echo('{} | Cif<{}> skipping: encountered an error retrieving cif data: {}'.format(
+                datetime.utcnow().isoformat(), source_id, exception.__class__.__name__))
         else:
             try:
                 if skip_partial_occupancies and cif.has_partial_occupancies():
-                    click.echo('{} | Cif<{}> skipped: contains partial occupancies'.format(
+                    click.echo('{} | Cif<{}> skipping: contains partial occupancies'.format(
                         datetime.utcnow().isoformat(), source_id))
                 else:
                     if not dry_run:
+
+                        if number_species is not None:
+                            cif._set_attr('number_species', number_species, clean=False)
+
                         batch.append(cif)
```

Comment:
This new attribute is slightly weird: it's not really a property of the CifData node but of the way it was obtained ... and might be in the end not related to the real number of species in the structure represented by the cif (if for instance the cif has an error).

Reply:
I agree that it is a little bit weird, but I think it is important nonetheless. Maybe in a different form. I am not suggesting that it says something about the contents of the `CifData` but exactly through what query it was extracted. Maybe it would be better to generalize this and provide a flag to store the query parameters in the `CifData` node as an attribute. The reason I needed this personally was because I query the databases in large chunks, only based on number of species, but I always store them in the same group, e.g. `cod_cif_raw`. However, later I would still need to be able to quickly get all the binaries, let's say. Without this attribute, it is pretty much impossible. Even if you parse it, some won't be parsable or give the wrong answer, where all I just really want to know is which cifs where returned by the original database query for that number of species.

Reply:
ok, but then, what about an extra, rather than an immutable attribute?

Reply:
Ok, you have convinced me. In any case, it probably deserves its own PR. I was trying to smuggle it in, which is not the correct approach. Let me remove it for now